rero_ils/modules/items/dumpers.py

```diff
@@ -82,19 +83,23 @@ class ClaimIssueNotificationDumper(InvenioRecordsDumper):
 
     def dump(self, record, data):
         """Dump an item issue for claim notification generation."""
-        assert record.is_issue, 'record must be an `ItemIssue` resource'
-        assert (holding := record.holding), \
-            'record must be related to an existing `Holding`'
+        if not record.is_issue:
+            raise TypeError('record must be an `ItemIssue` resource')
+        if not (holding := record.holding):
+            raise MissingDataException('item.holding')
```

Comment:
Why not using a KeyError exception?

Reply:
To be more `specific` in the error management ; additionally `MissingDataException` inherits from `KeyError`